agent/consul/server.go

```diff
@@ -1025,11 +1032,22 @@ func (s *Server) setupRaft() error {
 				return fmt.Errorf("recovery failed to parse peers.json: %v", err)
 			}
 
+			// It's safe to pass nil as the handle argument here because we won't call
```

Comment:
TIL that RecoverCluster specifically states that the FSM passed in is only used for snapshotting purposes and it should be discarded.